app/models/hanuman/observation.rb

```diff
@@ -121,6 +122,14 @@ def fill_answer
       end
     end
 
+    def edit_gnss
+      if new_record?
+        return
+      elsif location_metadata.present? && (latitude_changed? || longitude_changed? || speed_changed? || altitude_changed? || accuracy_changed? || direction_changed?)
+        update_column(:location_metadata, nil)
```

Comment:
This is a `before_save` callback, so you should be able to just set `self.location_metadata = nil` and the eventual save operation will deal with updating the database. One less query, and it avoid the possibility of a situation where the second DB transaction fails for some reason and the user loses their metadata but doesn't get the updates.

Reply:
To clarify, you are suggesting changing line 129 to `self.location_metadata = nil`?

Reply:
Yep!